File: i2f-extension/i2f-extension-swl/src/main/java/i2f/extension/swl/impl/sm/antherd/supplier/SwlAntherdSm3MessageDigesterSupplier.java
package i2f.extension.swl.impl.sm.antherd.supplier;

import i2f.extension.swl.impl.sm.antherd.SwlAntherdSm3MessageDigester;
import i2f.swl.std.ISwlMessageDigester;
import i2f.swl.std.supplier.ISwlMessageDigesterSupplier;

/**
 * @author Ice2Faith
 * @date 2026/4/8 15:34
 * @desc
 */
public class SwlAntherdSm3MessageDigesterSupplier implements ISwlMessageDigesterSupplier {
    @Override
    public ISwlMessageDigester get() {
        return new SwlAntherdSm3MessageDigester();
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/extension/impl/sm/antherd/supplier/SwlAntherdSm3MessageDigesterSupplier.js
import SwlAntherdSm3MessageDigester from "../SwlAntherdSm3MessageDigester";

/**
 * @return {SwlAntherdSm3MessageDigesterSupplier}
 * @implements ISwlMessageDigesterSupplier
 */
function SwlAntherdSm3MessageDigesterSupplier() {

}

/**
 * @return {ISwlMessageDigester}
 */
SwlAntherdSm3MessageDigesterSupplier.prototype.get = function () {
    return new SwlAntherdSm3MessageDigester()
}

export default SwlAntherdSm3MessageDigesterSupplier
